io/module_wrt_grid_comp.F90

```diff
@@ -1749,6 +1790,269 @@ subroutine recover_fields(file_bundle,rc)
+!
+   subroutine mask_fields(file_bundle,rc)
+
+     type(ESMF_FieldBundle), intent(in)              :: file_bundle
+     integer,                intent(out),   optional :: rc
+!
+     integer i,j,k,ifld,fieldCount,nstt,nend,fieldDimCount,gridDimCount
+     integer istart,iend,jstart,jend,kstart,kend,km
+     type(ESMF_Grid)  fieldGrid
+     type(ESMF_TypeKind_Flag) typekind
+     type(ESMF_TypeKind_Flag) attTypeKind
+     character(len=ESMF_MAXSTR) fieldName
+     type(ESMF_Field),   allocatable                  :: fcstField(:)
+     real(ESMF_KIND_R4), dimension(:,:),     pointer  :: var2dPtr2dr4
+     real(ESMF_KIND_R4), dimension(:,:,:),   pointer  :: var3dPtr3dr4
+     real(ESMF_KIND_R4), dimension(:,:,:),   pointer  :: vect3dPtr2dr4
+     real(ESMF_KIND_R4), dimension(:,:,:,:), pointer  :: vect4dPtr3dr4
+     real(ESMF_KIND_R4), dimension(:,:), allocatable  :: maskwrt
+
+     logical :: mvispresent=.false.
+     real(ESMF_KIND_R4) :: missing_value_r4=-1.e+10
+     real(ESMF_KIND_R8) :: missing_value_r8=9.99e20
+     character(len=ESMF_MAXSTR) :: msg
+
+     save maskwrt
+
+     call ESMF_LogWrite("call mask field on wrt comp",ESMF_LOGMSG_INFO,rc=RC)
+
+! get fieldCount
+     call ESMF_FieldBundleGet(file_bundle, fieldCount=fieldCount, &
+         grid=fieldGrid, rc=rc)
+     if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+        line=__LINE__, file=__FILE__)) return  ! bail out
+! get gridDimCount
+     call ESMF_GridGet(fieldgrid, dimCount=gridDimCount, rc=rc)
+     if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+        line=__LINE__, file=__FILE__)) return  ! bail out
+
+     allocate(fcstField(fieldCount))
+     call ESMF_LogWrite("call mask field get fcstField",ESMF_LOGMSG_INFO,rc=RC)
+     call ESMF_FieldBundleGet(file_bundle, fieldList=fcstField, itemorderflag=ESMF_ITEMORDER_ADDORDER, rc=rc)
+
+! generate the maskwrt according to surface pressure
+     if( first_getmaskwrt ) then
+
+     do ifld=1,fieldCount
+       !call ESMF_LogWrite("call mask field get fieldname, type dimcount",ESMF_LOGMSG_INFO,rc=RC)
+       call ESMF_FieldGet(fcstField(ifld),name=fieldName,typekind=typekind,dimCount=fieldDimCount, rc=rc)
+       !write(msg,*) 'fieldName,typekind,fieldDimCount=',trim(fieldName),typekind,fieldDimCount
+       !call ESMF_LogWrite("call mask field: "//trim(msg),ESMF_LOGMSG_INFO,rc=RC)
+       if (.not. allocated(maskwrt)) then
+         if ( typekind == ESMF_TYPEKIND_R4 .and. fieldDimCount == gridDimCount) then
+           call ESMF_FieldGet(fcstField(ifld),localDe=0, farrayPtr=var2dPtr2dr4, rc=rc)
+           istart = lbound(var2dPtr2dr4,1)
+           iend   = ubound(var2dPtr2dr4,1)
+           jstart = lbound(var2dPtr2dr4,2)
+           jend   = ubound(var2dPtr2dr4,2)
+           allocate(maskwrt(istart:iend,jstart:jend))
+           maskwrt(istart:iend,jstart:jend)=1.0
+         endif
+       endif
+       if(index(trim(fieldName),"pressfc")>0) then
```

Comment:
What will happen if "pressfc" is NOT present in the file_bundle? Looks like maskwrt will remain unallocated, yet it is used in the following do loop without checking its allocation status. Have you tested this code change in such case?

Reply:
I checked the code more carefully, maskwrt will be allocated but it will not be assigned value 0.0 to indicate missing value if "pressfc" is not found in the file_bundle.

Reply:
Hi Dusan,

Thanks for digging into the code! And yes, the assumption is that there must be a "pressfc" field in the output field bundle, otherwise the output fields will not be masked out properly for those points outside of the computation grid. But, in any case, the forecast model will still run, only with some output fields not being properly masked out. 

Appreciate it if you could provide some suggestions.

Thanks!

Bin